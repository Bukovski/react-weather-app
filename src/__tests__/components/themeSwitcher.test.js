import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import Temperature from "../../components/temperature.component";


const data = {
	"actualTemperature": "7",
	"maxTemperature": 8,
	"minTemperature": 10
}

describe("Temperature Component", () => {
	it('should render props', async () => {
		const { container } = render(<Temperature data={ data }/>)
		
		await waitFor(() => {
			expect(screen.queryByText('7')).toBeInTheDocument();
			expect(screen.queryByText('8')).toBeInTheDocument();
			expect(screen.queryByText('10')).toBeInTheDocument();
		}, { timeout: 1500 });
		
		expect(container.firstChild).toMatchSnapshot();
	});
});