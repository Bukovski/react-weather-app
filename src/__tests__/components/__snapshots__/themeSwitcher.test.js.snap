// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature Component should render props 1`] = `
<div
  class="temperature"
>
  <div
    class="temperature__deg"
  >
    <span>
      7
    </span>
  </div>
  <div
    class="temperature__wrap"
  >
    <div
      class="temperature__mesure"
    >
      °C
    </div>
    <div
      class="temperature__box"
    >
      <div
        class="temperature__max"
      >
        <span
          class="temperature__mark"
        >
          ↑
        </span>
        <p
          class="temperature__text"
        >
          <span>
            8
          </span>
        </p>
        <span
          class="temperature__unit"
        >
          °C
        </span>
      </div>
      <div
        class="temperature__min"
      >
        <span
          class="temperature__mark"
        >
          ↓
        </span>
        <p
          class="temperature__text"
        >
          <span>
            10
          </span>
        </p>
        <span
          class="temperature__unit"
        >
          °C
        </span>
      </div>
    </div>
  </div>
</div>
`;
